### PR TITLE
feat(3d-tiles): support s2 bounding volume

### PR DESCRIPTION
**Pull request by mspivak-actionengine**

### CODE REVIEW

File: modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import {getS2CellFromToken, get2dRegionFromS2Cell} from './s2-utils-ext';
```
Comment (ibgreen) [inline modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts:1]: The filename is unreadable. `convert-s2-to-obb.ts` ?

File: modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import {Vector3} from '@math.gl/core';
+import {OrientedBoundingBox, makeOrientedBoundingBoxFromPoints} from '@math.gl/culling';
+
+import {Ellipsoid} from '@math.gl/geospatial';
+
+export type S2BoundingVolume = {
```
Comment (ibgreen) [inline modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts:7]: Add at least one line JSDoc. Ideally one line JSDoc on every field in the type as well.

File: modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  token: string; // This can be an S2 key or token
+  minimumHeight: number;
+  maximumHeight: number;
+};
+
+function convertCartToXYZ(longitude, latitude, height, result?) {
```
Comment (ibgreen) [inline modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts:13]: - What is `Cart`? Please type out. `LngLat`? `Cartographic`?
- Add JSDoc line
- Add typescript types

File: modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  return new Vector3(point[0], point[1], point[2]);
+}
+
+/**
+ * Converts S2BoundingVolume to OrientedBoundingBox
+ * @param {S2BoundingVolume} s2bv - s2 bounding volume to convert
+ * @param {OrientedBoundingBox} [result] Optional object onto which to store the result.
+ * @returns The modified result parameter or a new `OrientedBoundingBox` instance if not provided.
+ */
+export function convertS2BVtoBox(
```
Comment (ibgreen) [inline modules/3d-tiles/src/lib/utils/s2/s2-bv-obb.ts:24]: Type out...
```suggestion
export function convertS2BoundingVolumeToOBB(
```

File: modules/3d-tiles/src/lib/utils/s2/s2-geometry.ts [hunk 5]
```diff
@@ -0,0 +1,208 @@
+/*
```
Comment (ibgreen) [inline modules/3d-tiles/src/lib/utils/s2/s2-geometry.ts:1]: Is this copied from deck.gl or is it an independent effort? Seeing this in multiple places, starting to feel like it is time to make an `@math.gl/s2` module. @Pessimistress what do you think about that?

**Review by ibgreen — COMMENTED** [review]
Nice work! 

- Please make a thorough pass on typing, comments, variable naming before landing, see inline comments.
- There are no tests, at least a minimal test case for converting an S2 cell to an OBB would make a big difference.

Remember that we write this code not primarily for ourselves, but to make it easy for the maintainers who come after us who have to try to understand we wrote. Anything we can do to improve readability, clarity, typing, comments, test cases will help the next reader.

I do think moving the s2 utils and the new s2toOBB function to math.gl could make a lot of sense. That would keep complexity down here.

File: modules/3d-tiles/src/lib/parsers/helpers/parse-3d-implicit-tiles.ts [hunk 6]
```diff
@@ -10,6 +13,32 @@ const SUBDIVISION_COUNT_MAP = {
...
+  let cTile;
+  if (curTile && curTile.boundingVolume && curTile.boundingVolume.box) {
+    if (typeof curTile.boundingVolume.cellId === 'undefined')
+      curTile.boundingVolume.cellId = getIdFromToken(curTile.boundingVolume.s2bv.token);
+    // Get box from childCellId
+    const childCellId = getS2ChildCellId(curTile.boundingVolume.cellId, index);
+    const childToken = getTokenFromId(childCellId);
+
+    // Clone object
+    const s2bv: S2BoundingVolume = JSON.parse(JSON.stringify(curTile.boundingVolume.s2bv));
```
Comment (belom88) [inline modules/3d-tiles/src/lib/parsers/helpers/parse-3d-implicit-tiles.ts:26]: The parse - stringify is not optimal approach. I suggest make a copy function. S2BoundingVolume object is not so long to do that.

**Review by ibgreen — APPROVED** [review]

**Review by belom88 — APPROVED** [review]

**Pull request merged**